packages/server/src/http/resolveHTTPResponse.ts

```diff
@@ -211,18 +321,53 @@ export async function resolveHTTPResponse<
       type: type,
       req,
     });
-    return endResponse(
-      {
-        error: getErrorShape({
-          config: router._def._config,
-          error,
-          type,
-          path: undefined,
-          input: undefined,
-          ctx,
-        }),
-      },
+
+    /**
+     * same as for a non-streamed response, do simple 2-step return
+     */
+
+    const untransformedJSON = {
+      error: getErrorShape({
+        config: router._def._config,
+        error,
+        type,
+        path: undefined,
+        input: undefined,
+        ctx,
+      }),
+    };
+
+    // yield header stuff
+    /**
+     * WARN: this can cause issues if streaming response has already yielded its headers,
+     * because this would be the second time we yield headers.
+     * For this condition to be realized, we need to be sure that `serialize` in
+     * `transformTRPCResponse` cannot throw.
+     */
```

Comment:
I think the two codepaths of streamed and non-streamed should be two distinctly different functions with their own error handling, we should never risk calling `initReponse` twice